lib/Sema/CSDiagnostics.cpp

```diff
@@ -3880,7 +3880,8 @@ bool NonOptionalUnwrapFailure::diagnoseAsError() {
     diagnostic = diag::invalid_force_unwrap;
 
   auto range = getSourceRange();
-  emitDiagnostic(diagnostic, BaseType).highlight(range).fixItRemove(range.End);
+  emitDiagnostic(diagnostic, resolveType(BaseType))
```

Comment:
Would it be possible to move resolveType to the constructor of this diagnostic instead?

Reply:
Sounds good, thank you!